src/AirBench/Controllers/BenchController.cs

```diff
@@ -19,30 +21,58 @@ public BenchController(IBenchRepository benchRepo, IReviewRepository reviewRepo)
         [AllowAnonymous]
         public ActionResult Index()
         {
-            var benches = benchRepo.List();
-            return View(benches);
+            return View();
         }
 
         public ActionResult Add()
         {
+            float latitude, longitude;
```

Comment:
I did this using javascript and cookies, but your way seems cleaner and easier to maintain.